x/claims/keeper/abci.go

```diff
@@ -120,20 +125,15 @@ func (k Keeper) ClawbackEmptyAccounts(ctx sdk.Context, claimsDenom string) {
 			return false
 		}
 
-		balances := k.bankKeeper.GetAllBalances(ctx, addr)
+		clawbackCoin := k.bankKeeper.GetBalance(ctx, addr, claimsDenom)
 
 		// prune empty accounts from the airdrop
-		if balances == nil || balances.IsZero() {
+		if clawbackCoin.IsZero() {
 			k.accountKeeper.RemoveAccount(ctx, acc)
```

Comment:
What if the account has a different coin balance? Why should it be deleted from the accounts state?

This is a weird side-effect for the claims-keeper to enact regardless